corehq/apps/sms/api.py

```diff
@@ -206,11 +212,44 @@ def send_message_via_backend(msg, backend=None, orig_phone_number=None):
         return False
 
 
+def random_password():
+    """
+    This method creates a random password for an sms user registered via sms
+    """
+    chars = string.ascii_uppercase + string.ascii_lowercase + string.digits
+    return ''.join(random.choice(chars) for x in range(15))
+
+
+def process_username(username, domain):
+    """
+    Loosely based on code from apps/users/forms.py:255
+    """
+    from corehq.apps.users.forms import validate_username
+
+    max_len_username = 80
+    if len(username) > max_len_username:
+        raise forms.ValidationError(
+            "Username %s is too long.  Must be under %d characters."
+            % (username, max_len_username))
+    # Check if the username contains invalid characters w/ django checker
+    validate_username('%s@commcarehq.org' % username)
```

Comment:
This line just checks for invalid characters. The next line appropriately formats it as username@domain.commcarehq.org (I pulled this code from: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/users/forms.py#L337)